torch/distributed/distributed_c10d.py

```diff
@@ -2378,11 +2387,18 @@ def barrier(group=GroupMember.WORLD,
     if _rank_not_in_group(group):
         return
 
+    opts = BarrierOptions()
+    if devicesId is not None:
+        try:
+            if type(devicesId) is not list: devicesId = [ devicesId ]
```

Comment:
If we don't enforce `list[init]` type, lets change this to an `isinstance` check.